mypy_django_plugin/transformers/models.py

```diff
@@ -254,56 +259,62 @@ def run_with_model_cls(self, model_cls: Type[Model]) -> None:
                 if manager_info is None:
                     continue
 
-            is_dynamically_generated = manager_info.metadata.get("django", {}).get("from_queryset_manager") is not None
-            if manager_name not in self.model_classdef.info.names or is_dynamically_generated:
-                manager_type = Instance(manager_info, [Instance(self.model_classdef.info, [])])
+            if manager_name not in self.model_classdef.info.names:
+                manager_type = self.model_parametrize(manager_info)
                 self.add_new_node_to_model_class(manager_name, manager_type)
-            elif self.has_any_parametrized_manager_as_base(manager_info):
-                # Ending up here could for instance be due to having a custom _Manager_
-                # that is not built from a custom QuerySet. Another example is a
-                # related manager.
-                custom_model_manager_name = manager.model.__name__ + "_" + manager_class_name
-                try:
-                    custom_manager_type = self.create_new_model_parametrized_manager(
-                        custom_model_manager_name, base_manager_info=manager_info
-                    )
-                except helpers.IncompleteDefnException:
-                    continue
+                continue
+            elif self._is_declared_inside_class(manager_name):
+                continue
+            elif not self._should_subclass_manager(manager_info):
+                self.add_new_node_to_model_class(manager_name, self.model_parametrize(manager_info))
```

Comment:
I think we should emit an error if a manager is explicitly typed with `Any`. I think this behaviour could be very surprising. I can easily see someone writing this and expecting a reusable manager:

```python
class MyManager(models.Manager[Any]):
    def has_items() -> bool:
        return self.exists()
```

When used that will result in this:

```python
class MyModel(models.Model):
    objects = MyManager()

reveal_type(MyModel.objects.get())  # note: Revealed type is "Any"
```

I don't really see any use case for this, so I think we should nudge people in the right direction with a useful message (maybe a hint if that's possible?). Maybe something like this:

> A manager typed with "Any" will not have types for the model it's used on. You might want to use a "TypeVar" bound to "Model" instead (ie. "TypeVar('T', bound=models.Model)")

Not super happy with that wording, but something like that would probably be useful

Reply:
Are you sure we should deny the use of `Any`? We then also have to be careful so the error isn't emitted for our own plugin insertions of `Any`, etc.

> ... I can easily see someone writing this and expecting a reusable manager ...

While that might be true, it also feels like someone misunderstanding generics, in general, with mypy.

Reply:
> Are you sure we should deny the use of `Any`? We then also have to be careful so the error isn't emitted for our own plugin insertions of `Any`, etc.

That should be a problem though? We just need to emit the error only for explicit `Any`

> > ... I can easily see someone writing this and expecting a reusable manager ...
> 
> 
> 
> While that might be true, it also feels like someone misunderstanding generics, in general, with mypy.

Yeah, but that's exactly who I'd target with an error (or as noted above ideally a note) nudging in the right direction.

Reply:
It is not a problem to emit an error on explicit `Any`, but I think that it is extremely rare use case. In general, any `mypy` output is considered a problem, while `Any` type arg is absolutely valid. I doubt that typing library plugin should teach users to use generics, it is up to documentation, isn't it? We are still doing many things to lower required knowledge: if I were designing this plugin initially, I'd never allow `class MyManager(models.Manager)` to be reparametrized, because it is easily expressible with TypeVar (and this is a common pattern).